### ISSUE TITLE
Non-otpauth URI QR Code Crashes FreeOTP

### ISSUE THREAD

**Issue by jciskey** [freeotp/freeotp-android#172]
Running the QR Code import on a QR code that isn't an otpauth URI will cause FreeOTP to crash without any explanation or warning.

Obviously this is an edge case, since most users will only be importing QR codes from well-validated sources that generate correct URIs. However, if there are multiple QR codes (some valid, some not), or if the QR code is coming from an experimental source, this could cause problems that are unexplainable by or to the user.

Attached is an example bad QR code that will crash FreeOTP. The encoded text value is: `2WCMLZQZ25YYQ6Q5`

![test-bad-qr](https://user-images.githubusercontent.com/5051960/34344397-233ffcf4-e9ab-11e7-81ea-246161aefd76.png)

**Comment by bostrt** [comment]
Hi @jciskey thanks for reporting this. I'm new to the FreeOTP community and taking a look at some outstanding issues, including this one. 

It looks like this issue was resolved in e1cfd62 but is not yet released. 

I'll run this through a test build to make sure its resolved and let you know.

**Comment by bostrt** [comment]
Looks like this is still an issue even with using a build with commit e1cfd62. I'll take a deeper look at this tomorrow. 

```
2020-07-30 22:12:02.962 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.Token.validateTokenURI(Token.java:143)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.Token.<init>(Token.java:66)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.Token.<init>(Token.java:215)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.Token.<init>(Token.java:219)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.add.ScanActivity.addTokenAndFinish(ScanActivity.java:72)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.add.ScanActivity.access$000(ScanActivity.java:50)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at org.fedorahosted.freeotp.add.ScanActivity$ScanBroadcastReceiver.onReceive(ScanActivity.java:60)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1555)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.os.Handler.handleCallback(Handler.java:883)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:100)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.os.Looper.loop(Looper.java:214)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:7356)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
2020-07-30 22:12:02.963 6878-6878/org.fedorahosted.freeotp W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
2020-07-30 22:12:02.965 6878-6878/org.fedorahosted.freeotp D/AndroidRuntime: Shutting down VM
2020-07-30 22:12:02.966 6878-6878/org.fedorahosted.freeotp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fedorahosted.freeotp, PID: 6878
    java.lang.RuntimeException: Error receiving broadcast Intent { act=org.fedorahosted.freeotp.ACTION_CODE_SCANNED flg=0x10 (has extras) } in org.fedorahosted.freeotp.add.ScanActivity$ScanBroadcastReceiver@53cf382
        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1565)
        at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.fedorahosted.freeotp.Token.getID()' on a null object reference
        at org.fedorahosted.freeotp.TokenPersistence.tokenExists(TokenPersistence.java:72)
        at org.fedorahosted.freeotp.add.ScanActivity.addTokenAndFinish(ScanActivity.java:81)
        at org.fedorahosted.freeotp.add.ScanActivity.access$000(ScanActivity.java:50)
        at org.fedorahosted.freeotp.add.ScanActivity$ScanBroadcastReceiver.onReceive(ScanActivity.java:60)
        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1555)
        at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2) 
        at android.os.Handler.handleCallback(Handler.java:883) 
        at android.os.Handler.dispatchMessage(Handler.java:100) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 

```

**Comment by bostrt** [comment]
e1cfd62 hasn't been merged yet but once it is we'll be able to close this out. There's still some new work being done and once its merged it will include e1cfd62. I'll leave this open for the time being.
